server/ladders-remote.ts

```diff
@@ -70,25 +70,46 @@ export class LadderStore {
 		const formatid = this.formatid;
 		const p1 = Users.getExact(p1name);
 		const p2 = Users.getExact(p2name);
-		room.update();
-		room.send(`||Ladder updating...`);
-		const [data, error] = await LoginServer.request('ladderupdate', {
+
+		const ladderUpdatePromise = LoginServer.request('ladderupdate', {
 			p1: p1name,
 			p2: p2name,
 			score: p1score,
 			format: formatid,
 		});
+
+		// calculate new Elo scores and display to room while loginserver updates the ladder
+		const [p1OldElo, p2OldElo] = (await Promise.all([this.getRating(p1!.id), this.getRating(p2!.id)])).map(Math.round);
+		const p1NewElo = Math.round(this.calculateElo(p1OldElo, p1score, p2OldElo));
+		const p2NewElo = Math.round(this.calculateElo(p2OldElo, 1 - p1score, p1OldElo));
+
+		room.update();
+		room.send(`||Ladder updating...`);
+
```

Comment:
```suggestion
```

These are no longer necessary now that the rest is synchronous.

Reply:
Updated.